core/src/test/scala/io/projectglow/gff/GffReaderSuite.scala

```diff
@@ -0,0 +1,280 @@
+        null,
+        null,
+        null,
+        "chromosome",
+        "genomic DNA",
+        null,
+        null,
+        null,
+        null
+      ),
+      Row(
+        "NC_000001.11",
+        "BestRefSeq",
+        "pseudogene",
+        11873,
+        14409,
+        null,
+        null,
+        null,
+        "gene-DDX11L1",
+        "DDX11L1",
+        null,
+        Array("GeneID:100287102", "HGNC:HGNC:37102").toSeq,
+        null,
+        null,
+        "DEAD/H-box helicase 11 like 1 (pseudogene)",
+        "Gene",
+        "DDX11L1",
+        "transcribed_pseudogene",
+        null,
+        null,
+        null,
+        null,
+        "true",
+        "passed",
+        null
+      )
+    )
+    assert(dfRows.sameElements(expectedRows))
+  }
+
+  test("Read gff with user-specified schema containing attributesField") {
+    val schemaWithAttributesField = StructType(
+      gffBaseSchema.fields ++ testOfficialFields ++ testUnofficialFields
+    )
+
+    val dfRow = spark
+      .read
+      .schema(schemaWithAttributesField)
+      .format(sourceName)
+      .load(testGff3)
+      .orderBy("seqId", "start")
+      .take(1)(0)
+
+    val expectedRow = Row(
+      "NC_000001.11",
+      "RefSeq",
+      "region",
+      0,
+      248956422,
+      null,
+      "+",
+      1,
+      "ID=NC_000001.11:1..248956422;Dbxref=taxon:9606,test;Name=1;chromosome=1;" +
+      "gbkey=Src;genome=chromosome;mol_type=genomic DNA;Is_circular=false",
+      "NC_000001.11:1..248956422",
+      "1",
+      null,
+      Array("taxon:9606", "test").toSeq,
+      false,
+      "1",
+      null,
+      "Src",
+      null,
+      null,
+      null,
+      "chromosome",
+      "genomic DNA",
+      null,
+      null,
+      null,
+      null
+    )
+    assert(dfRow == expectedRow)
+  }
+
+  // See the comment in io.projectglow.gff.GffResourceRelation.buildScan on use of filterFastaLines
+  test("df.count vs df.rdd.count") {
+    val df = spark
+      .read
+      .format(sourceName)
+      .load(testGff3)
+    assert(df.count() == 20)
+    assert(df.count() == df.rdd.count())
+  }
+
+  test("Read gff glob") {
+    val df = spark
+      .read
+      .format(sourceName)
+      .load(s"$testRoot/*")
+
+    assert(df.count() == 80)
+    assert(df.filter("start == 0").count() == 4)
+  }
+
+  test("Read empty gff") {
+    val df = spark
+      .read
+      .format(sourceName)
+      .load(testGff3Empty)
+
+    assert(df.schema == StructType(gffBaseSchema.fields.dropRight(1)))
+    assert(df.count() == 0) // requiredColumns will be empty
+    assert(df.rdd.count() == 0) // requiredColumns will be nonEmpty
+
+  }
+
+  test("GFF file format does not support writing") {
+    val df = spark
+      .read
+      .format(sourceName)
+      .load(testGff3)
+    val e = intercept[UnsupportedOperationException] {
+      df.write.format(sourceName).save("noop")
+    }
+    assert(e.getMessage.contains("GFF data source does not support writing!"))
+  }
+
```

Comment:
We should have a test for column pruning where we select only a few columns from the inferred schema.

Reply:
Added the "column pruning" test